### PR TITLE
feat: ensure traffic on the mesh is encrypted

### PR DESCRIPTION
**Pull request by XanderStrike**
### WHAT is this change about?

all unencrypted requests on the mesh will now be rejected

### Please provide any contextual information.

there are two exceptions:
  - capi requires an exception so that its db migrations in an init
  container can still access the db before the sidecar is up
  see: https://github.com/cloudfoundry/capi-k8s-release/issues/12
  - log cache requires an exception because it terminates its own tls in
  a way incompatible with the sidecars

### Have you read the [contributing guidelines](https://github.com/cloudfoundry/cf-for-k8s/blob/master/.github/contributing.md)?

- [x] YES
- [ ] NO

### Does this PR introduce a new ytt library?

- [ ] YES - please specify
- [x] NO

### Please provide Acceptance Criteria for this change?

1. Deploy cf for k8s
2. Enable docker support 
3. Push an app `cf push httpbin -o cfrouting/httpbin8080` 
4. Identify your service `kubectl get service -n cf-workloads -l cloudfoundry.org/app_guid=$(cf app httpbin --guid)` 
  * Looks like s-GUID 
5. SSH into one of your gateways 
  * `kubectl exec -it $(kubectl get pods -n istio-system | grep ingressgateway | head -n1 | awk '{print $1}') -n istio-system /bin/bash` 
6. Go to `/etc/certs` 
7. Send a plain text request to your app 
  * `curl http://<SERVICE NAME>.cf-workloads.svc.cluster.local:8080` 
  * It should fail with something like "Connectino reset by peer" 
  * NOTE: this doesn't work right now because we have to keep things in PERMISSIVE for CAPI to be HAPI 
8. Send an encrypted and authenticated request by specifying the cert, key, and rootca 
  * `curl -k --cacert root-cert.pem --key key.pem --cert cert-chain.pem https://<SERVICE NAME>.cf-workloads.svc.cluster.local:8080` 
  * See that it succeeds!"

### What is the level of urgency for publishing this change?

- [x] **Urgent** - unblocks current or future work
- [ ] **Slightly Less than Urgent**

### Tag your pair, your PM, and/or team!

@tcdowney @rodolfo2488 @rosenhouse @ndhanushkodi @keshav-pivotal

### CODE REVIEW

File: config/postgres.yml [hunk 1]
```diff
@@ -49,3 +49,17 @@ data:
...
+metadata:
+  name: "cf-db-allow-plaintext"
+  namespace: "cf-db"
+spec:
+  targets:
+  - name: cf-db-postgresql
+  - name: cf-db-postgresql-headless
+  peers:
+  - mtls:
+      mode: PERMISSIVE
```
Comment (cppforlife) [inline config/postgres.yml]: is this set to permissive because of the capi's init container?

Comment (XanderStrike) [reply config/postgres.yml]: Yes exactly.

Comment (cppforlife) [reply config/postgres.yml]: would you mind adding a comment in this file mentioning that?

File: config/postgres.yml [hunk 2]
```diff
@@ -49,3 +49,17 @@ data:
...
+metadata:
+  name: "cf-db-allow-plaintext"
+  namespace: "cf-db"
+spec:
+  targets:
+  - name: cf-db-postgresql
+  - name: cf-db-postgresql-headless
+  peers:
+  - mtls:
+      mode: PERMISSIVE # this allow's CAPI's init container to reach the database before the sidecar is up
```
Comment (cppforlife) [inline config/postgres.yml]: has to be '#!' instead of '# ' as ytt will complain about ambiguous comment

File: config/postgres.yml (hunk 1 above)
Comment (XanderStrike) [reply config/postgres.yml]: We've added comments to both the capi and log-cache workaround configuration.

File: config/postgres.yml (hunk 2 above)
Comment (XanderStrike) [reply config/postgres.yml]: We changed it after seeing your edit.

**Comment by jenspinney** [comment]
Looks good. Thanks!

**Comment by XanderStrike** [comment]
Thank you!

**Comment by pivotaljohn** [comment]
BTW, fantastic acceptance steps!!!  Thank you for taking the time to do so... and so clearly. 👍

**Pull request merged**